packages/main/src/utils/Strings.js

```diff
@@ -0,0 +1,66 @@
+/**
+	* Escape a regular expression text so that it can be used in a regular expression.
+	* @param {string} text The raw text
+	* @returns The regular expression text, with special characters having been escaped.
+	*/
+export function escapeRegex(text) {
+	return text.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+/**
+	* Replace all instances of a particular find text.˝
+	* @param {string} text The text to find/replace on
+	* @param {string} find The string to find
+	* @param {string|(s:string)=>string} replace The replace string or function
+	* @param {boolean=} caseInsensitive true if the replace is case insensitive, false by default
+	* @return {string} The string with all matches replaced.
+	*/
+export function replaceAll(text, find, replace, caseInsensitive) {
+	return text.replace(new RegExp(escapeRegex(find), `${caseInsensitive ? "i" : ""}g`), replace);
+}
+
+// Cache the regular expressions for repeated use
+const markupReplace = `&=&amp;,>=&gt;,<=&lt;,"=&quot;,'=&#39;`.split(",")
+	.map(p => p.split("="))
+	.map(([f, r]) => [new RegExp(escapeRegex(f), "g"), r]);
+
+/**
+	* Escape all html special characters in a string to their html entity equivalent.
+	* @param {string} text The raw text, which may contain markup
+	* @return {string} The escaped text
+	*/
+export function escapeHTML(text) {
```

Comment:
@d3xter666 ,
As defined, the text string input is a "plain text string" - it must not already contained escaped strings as that would lead to double escaping. All strings must be interpreted literally. It is not valid to assume that a raw HTML attribute would be necessary for the developer to define that property must escape the text assuming that raw attribute is interpreted as HTML markup.

I disagree fundamentally, if the user is passing a string it must not already be encoded as markup language with HTML entities.

For example -- let's suppose that I had an HTML like the following:
```html
<ui5-suggestion-item text="Dave & Busters"></ui5-suggestion-item>
```
Is it your expectation that everyone would need to make sure to already escape any strings that go into the `text` attribute and instead do the following?
```html
<ui5-suggestion-item text="Dave &amp; Busters"></ui5-suggestion-item>
```
That would be very weird to have the expectation that raw HTML attributes are meant to be interpreted as Markup. That's not how regular HTML tags work, so it should not be the case here either.

A developer who thought they should write the HTML in the second way should not be surprised that the suggestion item text shows up as `Dave &amp; Busters` as interpreted literally - since that is what the attribute is literally saying to do.

You can only know if the text that was passed in was NOT escaped, by detecting "dangerous characters" - but it is IMPOSSIBLE to know if a text was meant to be interpreted literally as special HTML entity characters or as the HTML equivalent.

It is best to just assume, as other native HTML elements do, that all HTML attributes are plain-text strings and are not meant to be interpreted as HTML strings across the board.

Reply:
So long story short, no one should be calling this function with any string that contains a string which is already escaped.